lib/redis_mutex.rb

```diff
@@ -56,6 +57,15 @@ def try_lock
     return false # Dammit, it seems that someone else was even faster than us to remove the expired lock!
   end
 
+  # Extends the expire time just in case process needs more time
+  def reset_expire
+    raise ExpireResetError, "extend_expire should be called only on locked resources" unless locked?
```

Comment:
This should check the ownership of the lock by `get == @expires_at.to_s` instead of just `locked?`